src/ansi-c/goto-conversion/goto_convert.cpp

```diff
@@ -1020,11 +1020,20 @@ void goto_convertt::convert_loop_contracts(
   {
     if(has_subexpr(invariant, ID_side_effect))
     {
-      throw incorrect_goto_program_exceptiont(
-        "Loop invariant is not side-effect free.", code.find_source_location());
+      if(check_side_effect)
+      {
+        throw incorrect_goto_program_exceptiont(
+          "Loop invariant is not side-effect free.",
+          code.find_source_location());
+      }
+      else
+      {
+        warning().source_location = code.find_source_location();
+        warning() << "Loop invariants is not side-effect-free." << eom;
```

Comment:
Nobody reads warnings. Can you describe the scenario where side effects would be ok?

Reply:
According to the offline discussion. I will move all the checks and throw to contracts component. They should not be in the converter.

Reply:
This PR now only allow loop contracts annotated to GOTO. Check for side effect will be in another PR.